techmatters/flex-plugins
# CHI1071 Webchat Pre-Engagement Mapping to Form Fields

Primary reviewer: @stephenhand 

## Description
- Populate data fields based on pre-engagement form in webchat widget to TabbedForms
- Changed `PrePopulateKeys` type to allow for customization based on pre-engagement fields and survey fields
- Supports input type, select type, and checkbox type fields

### Checklist
- [n/a] Feature flags added
- [n/a] Strings are localized
- [x] Tested for chat contacts (only webchat)
- [n/a] Tested for call contacts

### Related Issues
Fixes # [1071](https://tech-matters.atlassian.net/browse/CHI-1071)

### Verification steps
- Change configuration in webchat and form-definitions/tabbedForms to match a helpline with preengagement fields. ca-v1 is a good candidate. 
- Ensure PrepopulateKeys is filled out with some of the matching field names: example: ['age', 'province', ...]
- Reinstall dist in hrm-form-definitions
- In webchat widget, fill out the preengagement form, and accept a chat in Flex. 
- Check that the PrepopulateKeys included are filled out in Tabbed Forms
